sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1414,6 +1414,22 @@ object functions {
    */
   def md5(columnName: String): Column = md5(Column(columnName))
 
+  /**
+   * Calculates the SHA-2 family of hash functions and returns the value as a hex string.
+   *
+   * @group misc_funcs
+   * @since 1.5.0
+   */
+  def sha2(e: Column, bit: Int): Column = Sha2(e.expr, lit(bit).expr)
```

Comment:
Currently if given bit length is not one of the permitted values, the result value is NULL. So I think it should not be to fail fast?